### ISSUE TITLE
 [Hacktoberfest 2024] Refactor Word Scramble Game for Better Modularity

### ISSUE THREAD

**Issue by bijiyiqi2017** [bijiyiqi2017/dragon24#11]
### Issue Description:

**Overview:**  
We're looking for contributors to refactor the Word Scramble game code as part of Hacktoberfest 2024! The goal is to modularize the existing codebase by separating the logic, utility functions, and DOM-related code into different files. This will improve maintainability, testability, and make the code more structured.

**Task Details:**

1. **Refactor Utility Functions:**
   - Move the word scrambling function into a separate file (`utils.js`).
   - Ensure this module exports only the necessary functions.

2. **Separate Game Logic:**
   - Move the game logic (e.g., starting the game, checking guesses) into a new file (`game.js`).
   - Ensure the game logic interacts with the utility functions appropriately.

3. **Handle DOM Manipulation Separately:**
   - Move DOM-related code (e.g., updating the HTML elements, adding event listeners) into a separate file (`dom.js`).
   - Ensure DOM updates are handled independently of the game logic.

4. **Main Entry Point:**
   - Create an `index.js` file that ties everything together and initializes the game when the document is ready.

5. **Consider HTML and CSS Changes:**
   - As you refactor the JavaScript code, review and update the HTML and CSS as necessary to ensure they align with the new structure and functionality. This may include changes to class names, IDs, or the structure of the HTML elements to accommodate any new functionality.

**Example Refactor Structure:**
- `src/utils.js`: Utility functions (e.g., `scrambleWord`)
- `src/game.js`: Game logic (e.g., `startGame`, `checkGuess`)
- `src/dom.js`: DOM manipulation (e.g., `updateScrambledWord`, `addEventListeners`)
- `src/index.js`: Main entry point to initialize the game

**Resources:**
- [Current Game Code](https://github.com/bijiyiqi2017/dragon24/blob/1f73573890827891eb3740c634cb9cbcd44918f2/script.js)
- [Hacktoberfest 2024 Rules and Guidelines](https://hacktoberfest.digitalocean.com/)

**Suggested changes are outlined in this issue, but feel free to improve on the approach as long as the game logic remains intact.**

### Guidelines:
- Follow the repository's `CONTRIBUTING.md` for pull request guidelines.
- Make sure to write clean, readable, and modular code.
- Ensure your HTML and CSS structure supports the new JavaScript logic.
- Test your code to ensure the game works correctly after your changes.

### Acceptance Criteria:
- The code is properly split into modular files (`utils.js`, `game.js`, `dom.js`, `index.js`).
- The game works as expected without any regression.
- Ensure code adheres to best practices (use 4 spaces for indentation, follow existing styles, etc.).

### How to Test Your Code:
1. **Run the Game Locally:** After making your changes, run the game in your local development environment to ensure everything functions as expected.
2. **Check for Errors:** Open the browser console to check for any errors or warnings while running the game.
3. **Verify Functionality:** Test all aspects of the game, including scrambling words, checking guesses, and DOM interactions.

### How to Contribute:
1. Fork the repository.
2. Create a branch specific to this issue (e.g., `refactor-game`).
3. Make the necessary changes as outlined above.
4. Test your changes thoroughly to ensure everything works correctly.
5. Submit a Pull Request, ensuring you mention this issue in the description.

### Submitting a Pull Request:
1. **Go to Your Forked Repository on GitHub:**
   - Navigate to your GitHub account and find the repository you forked. You can find it in your profile under "Repositories."

2. **Click on the “Pull Requests” Tab:**
   - At the top of the repository page, you will see several tabs (Code, Issues, Pull Requests, etc.). Click on the “Pull Requests” tab to view existing pull requests and create a new one.

3. **Click on the “New Pull Request” Button:**
   - On the Pull Requests page, you will see a green button that says “New Pull Request.” Click this button to start the process of creating a new pull request.

4. **Select Your Branch from the Dropdown Menu:**
   - In the “Compare” section, you will see a dropdown menu where you can select the branch you created for your changes (e.g., `refactor-game`). Make sure you select the correct branch.

5. **Write a Title and Description for Your Pull Request:**
   - In the title field, provide a concise and descriptive title for your pull request (e.g., “Refactor Word Scramble Game for Better Modularity”).
   - In the description field, explain the changes you made, how they improve the project, and any relevant details that reviewers should know. Be sure to mention this issue number (e.g., “This PR addresses issue #XX”).

6. **Review the Changes:**
   - Before submitting, GitHub will display a comparison of changes between your branch and the base branch (usually `main` or `master`). Review this to ensure all intended changes are included.

7. **Click “Create Pull Request”:**
   - Once you’re satisfied with your title, description, and the changes, click the green “Create Pull Request” button at the bottom of the page.

8. **Follow Up:**
   - After creating your pull request, monitor the conversation on the pull request page. Be prepared to answer any questions from maintainers or make additional changes if requested.

9. **Stay Engaged:**
   - Engage with any feedback or comments on your pull request. Respond to suggestions or requests for changes to help get your PR merged.

### Hacktoberfest 2024:
This issue is open to Hacktoberfest participants! Once your PR is approved and accepted via Hacktoberfest 2024, it will count toward your Hacktoberfest contributions.

## ⭐ How to Star This Repository:

Starring a repository on GitHub is a simple way to show your appreciation and help others discover it. Here’s how to do it:

1. **Visit the Repository:** Go to the main page of the repository you want to star. You can find it through a link or by searching on GitHub.

2. **Locate the Star Button:** At the top right corner of the repository page, you’ll see a button that says “Star.” It’s usually located next to the “Fork” button.

3. **Click the Star Button:** Simply click the “Star” button. Once you do, it will change to show that you’ve starred the repository (it will display “Unstar” if you want to remove your star later).

4. **Optional - Explore Starring Benefits:** By starring repositories, you can easily find them later in your GitHub profile under the “Stars” tab. It also helps other users see which projects are popular and worth checking out.

Thank you for supporting this project!
